src/main/java/org/wso2/config/mapper/TomlParser.java

```diff
@@ -128,14 +128,26 @@ private static Map<String, String> processSecrets(TomlParseResult result) {
         TomlTable table = result.getTable(ConfigConstants.SECRET_PROPERTY_MAP_NAME);
         TomlTable runtimeSecretsConfig = result.getTable(ConfigConstants.RUNTIME_SECRETS);
         if (table != null) {
-            table.dottedKeySet().forEach(key -> context.put(key, table.getString(key)));
+            table.dottedKeySet().forEach(key -> context.put(filterDottedContent(key), table.getString(key)));
         }
         if (runtimeSecretsConfig != null) {
             runtimeSecretsConfig.dottedKeySet().forEach(key -> context.put(key, runtimeSecretsConfig.getString(key)));
         }
         return context;
     }
 
+    /**
+     * Extract key surrounded by double quotes.
+     * @param key
+     * @return key value without surrounding double quotes
+     */
+    static String filterDottedContent(String key) {
```

Comment:
Shall we rename this method as `extractQuotedContent`, since that would be a more meaningful name for its functionality?

Reply:
Updated the method name.